malio-processor/src/main/java/com/github/nalukit/malio/processor/constraints/generator/ValidatorGenerator.java

```diff
@@ -91,22 +92,34 @@ private CodeBlock createCheckMethodForSubValidator(ValidatorModel model) {
                                         ClassName.get(Objects.class),
                                         this.processorUtils.createGetMethodName(model.getFieldName()));
     switch (model.getType()) {
-      case NATIVE:
-        String vaidatorClassName = model.getSimpleClassName() + model.getPostFix();
-        builder.add("$T.INSTANCE.check(bean.$L());",
-                                        ClassName.get(model.getPackageName(),
-                                                      vaidatorClassName),
-                                        this.processorUtils.createGetMethodName(model.getFieldName()));
+      case ARRAY:
+        String vaidatorClassNameArray = model.getPackageName() + "." + model.getSimpleClassName() + model.getPostFix();
```

Comment:
This will be only generated for complex types annotated with @MalioValidator? Tests case for an Integer-Array?